src/text-buffer.coffee

```diff
@@ -18,8 +19,11 @@ class TextBuffer
   getLinesLayer: ->
     @linesLayer ?= new Layer(@getCharactersLayer(), new LinesTransform)
 
-  buildTabsLayer: (tabLength) ->
-    new Layer(@getLinesLayer(), new TabsTransform(tabLength))
+  buildHardTabsLayer: (tabLength) ->
+    new Layer(@getLinesLayer(), new HardTabsTransform(tabLength))
```

Comment:
I agree that we can wait on them sharing any code. My main concern was that the soft tabs layer be built on top of the hard tabs layer by default, but perhaps your larger point about calling a `.transformWith` on each layer as needed obviates this concern.